cppapi/client/zmqeventconsumer.cpp

```diff
@@ -1075,8 +1075,17 @@ bool ZmqEventConsumer::process_ctrl(zmq::message_t &received_ctrl,zmq::pollitem_
 
         case ZMQ_DELAY_EVENT:
         {
-            old_poll_nb = poll_nb;
-            poll_nb = 1;
+            // If poll_nb == 1, then we are already in a situation where events are being delayed
```

Comment:
Good point! I agree with you it would make sense to count the number of received ZMQ_DELAY_EVENTS and really _release_ once we have received as many ZMQ_RELEASE_EVENTS as ZMQ_DELAY_EVENTS.
There might be other issues if we don't do that

Reply:
@mliszcz , I implemented this counter in a5d2c80cfd7ac82f7b84e190cef4423babea23c5
In cppTango >= 9.4, we could eventually make this variable a private member of ZmqEventConsumer class. 
I didn't do it here to preserve the binary compatibility.
For the moment, this variable is useful/used only in ZmqEventConsumer::process_ctrl() method and the ZMQEventConsumer object is a singleton so it should be fine like this.
Could you please have a look?

Reply:
Thanks! This is exactly what I had in mind.